### PR TITLE
Optimize b16

### PR DESCRIPTION
**Pull request by sungeunk**
Ticket: 80756

### CODE REVIEW

File: src/plugins/intel_gpu/src/graph/layout_optimizer.cpp [hunk 1]
```diff
@@ -423,6 +427,9 @@ bool layout_optimizer::can_fuse_reorder_to_prev(program_node& prev, program_node
         if (prev.is_type<eltwise>() &&
             is_mixed_layout(prev, *next, false, {{ format::bs_fs_zyx_bsv32_fsv32, format::bs_fs_zyx_bsv32_fsv16 }}))
             return true;
+        if (prev.is_type<eltwise>() && next->is_type<convolution>()) {
```
Comment (sungeunk) [inline src/plugins/intel_gpu/src/graph/layout_optimizer.cpp:430]: remove

File: src/plugins/intel_gpu/src/kernel_selector/core/cl_kernels/include/batch_headers/fetch_data.cl [hunk 2]
```diff
@@ -857,6 +873,22 @@ inline uint get_bs_fs_zyx_bsv_fsv_index(uint b, uint f,  uint z, uint y, uint x,
         CAT(prefix, _PAD_BEFORE_SIZE_X),                            \
         CAT(prefix, _PAD_AFTER_SIZE_X), 16, 32)
 
+#define GET_DATA_BS_FS_YX_BSV16_FSV32_INDEX(prefix, b, f, y, x) \
```
Comment (isanghao) [inline src/plugins/intel_gpu/src/kernel_selector/core/cl_kernels/include/batch_headers/fetch_data.cl:876]: why is it defined twice?

File: src/plugins/intel_gpu/src/graph/layout_optimizer.cpp [hunk 3]
```diff
@@ -1117,32 +1119,20 @@ layout layout_optimizer::get_expected_layout(layout const& current_layout,
...
+        expected_format = node.get_required_output();
     } else if (_optimization_attributes.b_fs_zyx_fsv16_network &&
         deconvolution_b_fs_zyx_fsv16_opt(current_layout, output_or_weights_layout, prim)) {
         if ((current_layout.data_type == data_types::f32 && expected_tensor.batch[0] % 16 == 0) ||
             (current_layout.data_type == data_types::f16 && expected_tensor.batch[0] % 32 == 0))
             expected_format = cldnn::format::bs_fs_zyx_bsv16_fsv16;
         else
             expected_format = cldnn::format::b_fs_zyx_fsv16;
-    } else if (_optimization_attributes.b_fs_yx_fsv16_network &&
+    } else if ((_optimization_attributes.b_fs_yx_fsv16_network || device_info.supports_immad) &&
```
Comment (isanghao) [inline src/plugins/intel_gpu/src/graph/layout_optimizer.cpp:1135]: Do we need to have this workaround here?

File: src/plugins/intel_gpu/src/graph/layout_optimizer.cpp [hunk 4]
```diff
@@ -1472,7 +1465,8 @@ impl_types layout_optimizer::get_preferred_impl_type(program_node& node, format
         if (node.is_type<pooling>()) {
             auto pool_layout = node.get_output_layout();
             if (pool_layout.data_type == data_types::i8) {
-                if (pool_layout.format == format::b_fs_zyx_fsv32 || pool_layout.format == format::bs_fs_zyx_bsv32_fsv32)
+                if (pool_layout.format == format::b_fs_zyx_fsv32 || pool_layout.format == format::bs_fs_zyx_bsv16_fsv32 ||
+                    pool_layout.format == format::bs_fs_zyx_bsv32_fsv32)
```
Comment (isanghao) [inline src/plugins/intel_gpu/src/graph/layout_optimizer.cpp:1469]: Does it give performance benefit to use cldnn pooling for `zyx_bsv32_fsv32`?

**Review by isanghao — APPROVED** [review]

File: src/plugins/intel_gpu/src/graph/layout_optimizer.cpp [hunk 5]
```diff
@@ -805,14 +809,12 @@ static bool is_node_for_onednn(deconvolution_node const& node) {
         return false;
 
     // Onednn deconv does not support cross-precision
```
Comment (isanghao) [inline src/plugins/intel_gpu/src/graph/layout_optimizer.cpp:811]: Please remove old comment

**Pull request merged**